### PR TITLE
send cloudProvider to Clouddriver instead of providerType

### PR DESCRIPTION
**Pull request by anotherchrisberry**
depends on https://github.com/spinnaker/clouddriver/pull/1268

### CODE REVIEW

**Comment by robzienert** [comment]
ping. Would like this to go in if possible.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/TargetReferenceConfiguration.groovy [hunk 1]
```diff
@@ -31,7 +31,8 @@ class TargetReferenceConfiguration {
   String credentials
   List<String> regions
   List<String> zones
-  String providerType = "aws"
+  @Deprecated String providerType = "aws"
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/TargetReferenceConfiguration.groovy]: This seems like something we could do a better job of ... maybe get rid of the `providerType` attribute but leave a `setProviderType` (that sets a value on `cloudProvider`) if there is something deserializing to this.

**Comment by anotherchrisberry** [comment]
Updated to satisfy the @ajordens monster

**Comment by ajordens** [comment]
👍 🌮 🇨🇦

**Comment by anotherchrisberry** [comment]
let's find out!

**Pull request merged**